wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java

```diff
@@ -258,19 +284,89 @@ public void read(HttpServiceRequest request, HttpServiceResponder responder,
+	Properties properties = config.get();
+	if (!format.equalsIgnoreCase(Formats.AVRO) && !format.equalsIgnoreCase("binary")) {
+		outputFields = new ArrayList<>();
+		outputFields.add("body");
+		outputSchema = Format.TEXT.getSchema().toString();
+		return;
+	} else if ((format.equalsIgnoreCase(Formats.AVRO) && !properties.containsKey(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name())) || format.equalsIgnoreCase("binary")){
+		outputFields = new ArrayList<>();
+		outputFields.add("body");
+		outputSchema = Format.BLOB.getSchema().toString();
+		return;
+	}
+    String schemaName = properties.getProperty(PropertyIds.SCHEMA_NAME);
+    if (schemaName == null || schemaName.trim().isEmpty()) {
+      throw new IllegalArgumentException("Empty schema name");
+    }
+
+    String schemaRegistryUrl = properties.getProperty(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name());
+    LOG.error("Schema Registry Url: {}", schemaRegistryUrl);
+    if (schemaRegistryUrl == null || schemaRegistryUrl.trim().isEmpty()) {
+      throw new IllegalArgumentException("Empty schema registry url");
+    }
+
+    schemaRegistryUrl = schemaRegistryUrl.trim();
+    if (!schemaRegistryUrl.endsWith("/")) {
+      schemaRegistryUrl = schemaRegistryUrl + "/";
+    }
+
+    schemaRegistryUrl = schemaRegistryUrl + "schemaregistry/schemas/" + schemaName + "/versions/latest";
+
+    String schemaString = "";
+    String keytabLocation = config.getKeytabLocation();
+    String principal = config.getPrincipal();
+    if(Strings.isNullOrEmpty(keytabLocation) || Strings.isNullOrEmpty(principal)) {
+       schemaString = URLExtractor.extract(schemaRegistryUrl, 30);
+    } else {
+       schemaString = KerberosHttpClient.sendKerberisedGetRequest(schemaRegistryUrl,
+    		   principal, keytabLocation);
+    }
+    parseSchemaString(schemaString);
+  }
 
+  private void parseSchemaString(String schemaString) {
+	    outputFields = new ArrayList<>();
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("type", "record");
+        jsonObject.addProperty("name", "etlSchemaBody");
+        JsonArray fields = new JsonArray();
+	    Config config = ConfigFactory.parseString(schemaString);
+	    if (config.hasPath("schemaText")) {
+	      try {
+	        List<Schema.Field> schemaFields = Schema.parseJson(config.getString("schemaText")).getFields();
+	        for (Schema.Field schemaField: schemaFields) {
+	        	outputFields.add(schemaField.getName());
+	            JsonObject column = new JsonObject();
+	            column.addProperty("name", schemaField.getName());
+	            column.addProperty("type", "string");
+	        	fields.add(column);
+	        }
+	        jsonObject.add("fields", fields);
+	        outputSchema = jsonObject.toString();
+	      } catch (IOException e) {
+	        LOG.error(e.getMessage(), e);
+	      }
+	    } else {
+	      LOG.error("'schemaText' is not present in schema json.");
```

Comment:
Please check if can throw can exception from here so that error can be propagated to UI else change the log level of this message to Warn/Debug.